java/pom.xml

```diff
@@ -11,8 +11,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <arrow.version>3.0.0</arrow.version>
-        <dep.guava.version>20.0</dep.guava.version>
+        <arrow.version>16.1.0-20240904212746-09241242ea-dremio</arrow.version>
```

Comment:
Should we leave a public version of arrow? This is kind of an early preview one?

Reply:
This is the version that dremio is built against. It's available in the oss version, but point taken, we should be able to update to 17.0.0 release.